app/src/main/java/com/concordium/wallet/ui/passphrase/recover/PassPhraseRecoverInputFragment.kt

```diff
@@ -54,8 +54,8 @@ class PassPhraseRecoverInputFragment : Fragment() {
     override fun onResume() {
         super.onResume()
         (activity as BaseActivity).setActionBarTitle(R.string.pass_phrase_recover_input_title)
-        val inputMethodManager = context?.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-        inputMethodManager.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0)
+        val inputMethodManager = requireContext().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
```

Comment:
Is INPUT_METHOD_SERVICE always InputMethodManager? If not the casting is unsafe and I suggest changing it the following:

`val inputMethodManager = requireContext().getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager`

Then you can check if inputMethodManager is null before calling toggleSoftInput, so you avoid nullreference exception

Reply:
requireContext().getSystemService returns the handle to a system-level service by name. The class of the returned object varies by the requested name. Here the requested name is INPUT_METHOD_SERVICE ("input_method")
which is an InputMethodManager for management of input methods.